### ISSUE TITLE
One resolution too much

### ISSUE THREAD

**Issue by regiskuckaertz** [scala-open-letter/scala-open-letter.github.io#292]
With all due respect, I would like to suggest you either remove or rephrase this:

> We will not use or promote any software artifacts that are maintained by Mr. Pretty.

The technical output of Mr Pretty has nothing to do with his alleged social behaviour. Plus, this would be crippling to a huge portion of the downstream ecosystem, good developers who have no reason to be affected by this.

**Comment by bmc** [comment]
The original 23 signatories worked on and debated this letter for weeks before agreeing on that wording. At this point, now that the letter is public, more than 300 people have signed it, and there are more pending that have not yet been merged in.

We are not changing the wording of a document more than 300 people have already signed.

If you feel you cannot sign the document as it currently exists, then the answer is simple: Don't sign it.

We will not be changing the document.

Closing this issue.

**Comment by SethTisue** [comment]
> The technical output of Mr Pretty has nothing to do with his alleged social behaviour

Strongly disagree. As Yifan's letters makes abundantly clear, abusers exploit their status in the community to their advantage, and open source work is one of the ways that status is created and maintained.

> Plus, this would be crippling to a huge portion of the downstream ecosystem, good developers who have no reason to be affected by this

This can be addressed two ways:

* By acknowledging that the wordings in the letter are to be taken within reason. So for example, suppose I have a Scala job working on an application that depends heavily on Magnolia. If I sign the letter, am I committing to quit my job that very same day? No, but as I read the letter, I would be promising to work towards finding alternatives as soon as reasonably possible, within the usual practical constraints that we all face.
* By working towards building and promoting alternative technologies.

It is simply untrue that moving off of Jon's libraries tools is going to be “crippling” to the Scala ecosystem.

**Comment by bmc** [comment]
Thanks, @SethTisue, for the clarification. I clearly should not be responding to issues before I've had any coffee.

Though we're still not changing the letter's wording. ;-)

**Comment by regiskuckaertz** [comment]
Thank you @SethTisue and @bmc for taking the time to lay out your thoughts about this. I understand where you are coming from and I respect your decision.

> open source work is one of the major ways that status is created and maintained

Certainly. My understanding of the letter is you are against people _using_ their social status for predatory behaviour, not against people _acquiring_ social status through community effort. It is not very hard to find people in the history of our industry who have advanced the field in significant ways (and we make a living thanks in part to them) and yet sometime in their life they did things way worse than the allegations.

> By working towards building and promoting alternative technologies.

There is a problem with that: people change over time and so does their behaviour. For example, say you publish a library that is more elegant, more performant than Magnolia, it is a brilliant piece of engineering: mission accomplished. Then later, the vagaries of life turn you into a crypto-bro that damages the environment by mining bitcoin; perhaps you need the money to provide for your family. A lot of people will chastise you for it, for good reason, and distance themselves from you. According to that logic, a new letter will be published, encouraging people to build a better Magnolia yet again. The scenario where the library you create is worse on every level is even more perplexing.

I chose to sign the letter, because I empathise with the trauma experienced by these young women in many ways, but I will not abide by that last resolution.

**Comment by bmc** [comment]
I will not use anything Jon Pretty authors. I deleted any repos and presentations of mine that referred to his stuff (and that includes a presentation to our local Scala user group on _contextual_). That said, I don't use any of his stuff in anything I've written, so it is very easy for me to avoid his body of work. I understand that it is not so easy for others to avoid his work.

Still, I'm with Seth on this one: If I _were_ using something of his (or working somewhere that did), I would work toward replacing those artifacts with something else. To my knowledge, there are alternatives to pretty much everything he has written. Maybe some of those alternatives seem less elegant, but they exist.

As for Emacs, yeah, I still use it sometimes, too. And I also signed the anti-RMS letter. I acknowledge the cognitive dissonance of that situation.
